### ISSUE TITLE
SEACAS: Fix Error in new GCC 8.3.0 Werror Build

### ISSUE THREAD

**Issue by ZUUL42** [trilinos/Trilinos#6605]
## Bug Report
@trilinos/seacas

### Description
When testing the new GCC 8.3.0 build w/Werror on, #6295, there is a error coming from SEACAS that prevents turning on Werror globally.
See recent test build here: [CDash](https://testing.sandia.gov/cdash/index.php?project=Trilinos&display=project&filtercount=3&showfilters=1&filtercombine=and&field1=site&compare1=61&value1=ascic166&field2=buildname&compare2=61&value2=PR-gcc_8.3-test-Trilinos_pullrequest_gcc_8.3.0-42&field3=buildstamp&compare3=61&value3=20200115-1644-Experimental)

Werror flags used:
`set(CMAKE_CXX_FLAGS "-std=c++11 -fno-strict-aliasing -Wall -Wno-clobbered -Wno-vla -Wno-pragmas -Wno-unknown-pragmas -Wno-parentheses -Wno-unused-local-typedefs -Wno-literal-suffix -Wno-deprecated-declarations -Wno-misleading-indentation -Wno-int-in-bool-context -Wno-maybe-uninitialized -Wno-cast-function-type -Wno-class-memaccess -Wno-inline -Wno-nonnull-compare -Wno-address -DTRILINOS_HIDE_DEPRECATED_HEADER_WARNINGS -Werror" CACHE STRING "Warnings as errors settings")`
[Diff](https://github.com/trilinos/Trilinos/compare/develop...ZUUL42:gcc_8.3)

**Comment by lucbv** [comment]
@gsjaardema it would be nice to have that issue looked at since it is one of the last two issues holding up the gcc/8.3.0 build promotion to the PR auto-tester.

**Comment by gsjaardema** [comment]
What branch/tag is being used for this compilation.  It is mentioning `strncpy` and `CM_CAST` in the warning/error messsage, but `CM_CAST` does not exist in any SEACAS source code and there is no `strncpy` in `ExoIICTests.cxx` that I can find.

**Comment by gsjaardema** [comment]
OK, the `CM_CAST` comes from some processing that CMake does, but there is no `strncpy` in the processed file that I can see so still confused as to what branch/tag is being used.

**Comment by gsjaardema** [comment]
See https://gitlab.kitware.com/cmake/cmake/issues/18038  This is an error in cmake itself and not SEACAS.  It looks like it has been fixed in https://gitlab.kitware.com/cmake/cmake/commit/3af0438b4bccc599792f61a9f4be81b4bb36c3e3

Looks like we need a newer version of CMake to fix this.

**Comment by lucbv** [comment]
@gsjaardema thanks for catching this cmake issue!
It looks like we are at the mercy of some funny business in the auto-tester logic.
It seems that the modules are defined twice (using the gcc/7.2.0 PR tester to illustrate this):
* first [here](https://github.com/trilinos/Trilinos/blob/c962db7665df7f841f62de8f3f7058e0ba6c3449/cmake/std/PullRequestLinuxDriverTest.py#L135) 
* second [here](https://github.com/trilinos/Trilinos/blob/c962db7665df7f841f62de8f3f7058e0ba6c3449/cmake/std/sems/PullRequestGCC7.2.0TestingEnv.sh)
We probably need to synchronize these, but even better would be to have the modules defined only once. 
@prwolfe @william76 do you know how the python and cmake scripts mentioned above interact?

**Comment by prwolfe** [comment]
Luc,

The python controls the actual PR testing.  The shell scripts were only left in place to facilitate developers re-creating PR builds as needed.  I have been meaning to consolidate those to use a single source but have not got to that point.

Paul

**Comment by ZUUL42** [comment]
Update to cmake and ninja versions has allowed werror to be set globally.
Closing
